src/Workspaces/Core/Portable/SourceGeneration/SourceGeneratorPresence.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.SourceGeneration;
+
+/// <summary>
+/// Indicates the presence and types of source generators in a project.
+/// </summary>
+internal enum SourceGeneratorPresence
+{
+    /// <summary>
+    /// The project contains no source generators.
+    /// </summary>
+    NoSourceGenerators,
```

Comment:
practically, i assume we're at the point that this never happens.  So you could really just have this ContainsRequiredSourceGenerators and DoesNotContainRequiredSourceGenerators (just as a named bool) right?

Note: i'm not blocking this.  if you feel this is clear and helpful, then we can keep.  I'm just thinking out loud and trying to determine if it really would matter in practice.